go/enclave/enclave_admin_service.go

```diff
@@ -514,20 +513,63 @@ func (e *enclaveAdminService) ingestL1Block(ctx context.Context, processed *comm
 		return nil, nil, err
 	}
 
-	rollupMetadata, err := e.rollupConsumer.ProcessRollups(ctx, rollups)
-	if err != nil && !errors.Is(err, components.ErrDuplicateRollup) {
-		e.logger.Error("Encountered error while processing l1 block rollups", log.ErrKey, err)
-		// Unsure what to do here; block has been stored
+	var rollupMetadataList []common.ExtRollupMetadata
+	if processed.HasEvents(common.RollupTx) {
+		rollupMetadataList, err = e.processRollups(ctx, processed, rollupMetadataList)
+		if err != nil {
+			return nil, nil, err
+		}
 	}
 
+	// Handle any L1 fork events
 	if ingestion.IsFork() {
 		e.registry.OnL1Reorg(ingestion)
-		err := e.service.OnL1Fork(ctx, ingestion.ChainFork)
-		if err != nil {
+		if err := e.service.OnL1Fork(ctx, ingestion.ChainFork); err != nil {
 			return nil, nil, err
 		}
 	}
-	return ingestion, rollupMetadata, nil
+	return ingestion, rollupMetadataList, nil
+}
+
+func (e *enclaveAdminService) processRollups(ctx context.Context, processed *common.ProcessedL1Data, rollupMetadataList []common.ExtRollupMetadata) ([]common.ExtRollupMetadata, error) {
+	rollupTxs := processed.GetEvents(common.RollupTx)
+	txsSeen := make(map[gethcommon.Hash]bool)
+
+	// verify and process each rollup one by one
+	for _, rollupTx := range rollupTxs {
+		extRollup, err := e.rollupConsumer.VerifyRollupData(rollupTx)
+		if err != nil {
```

Comment:
doesn't feel like error handling is right here. If I'm not mistaken, any error will stop the network, which is not what we want.
I'd say we should just log and skip after any error at this level